xunit.runner.wpf/Impl/RemoteTestUtil.Connection.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Threading;
+using xunit.runner.data;
+using xunit.runner.wpf.ViewModel;
+
+namespace xunit.runner.wpf.Impl
+{
+    internal sealed partial class RemoteTestUtil : ITestUtil
+    {
+        private sealed class Connection : IDisposable
+        {
+            private NamedPipeClientStream _stream;
+            private Process _process;
+            private ClientReader _reader;
+
+            internal NamedPipeClientStream Stream => _stream;
+
+            internal ClientReader Reader => _reader;
+
+            internal Connection(NamedPipeClientStream stream, Process process)
+            {
+                _stream = stream;
+                _process = process;
+                _reader = new ClientReader(stream);
+            }
+
+            internal void Close()
+            {
+                if (_process != null)
+                {
+                    Debug.Assert(_stream != null);
+
+                    try
+                    {
+                        _stream.WriteAsync(new byte[] { 0 }, 0, 1);
```

Comment:
One question is why we're using named pipes at all, instead of another pattern.  The common ones I see are:

1. Stdin/Stdout
2. Http with json payloads.

I'm sure that neither would be as efficient, but they are both more widely understood patterns.